osu.Game/Screens/Edit/EditorClock.cs

```diff
@@ -118,6 +118,7 @@ private void seek(int direction, bool snapped, double amount = 1)
 
             if (!snapped || ControlPointInfo.TimingPoints.Count == 0)
             {
+                seekTime = Math.Clamp(seekTime, 0, TrackLength);
```

Comment:
This clamp is done again lower down again in line 155. Putting these all over feels like playing whack-a-mole to a degree. Maybe it would be a good idea to move this clamp down lower in the call stack (like somewhere in `Seek()`/`transformSeek()`) so that it gets called on all possible paths?